test/remote/gateways/remote_simetrik_test.rb

```diff
@@ -13,51 +12,203 @@ def setup
       year: 2022,
       verification_value: '111'
     )
+    @credit_card_invalid = CreditCard.new(
+      first_name: 'Joe',
+      last_name: 'Doe',
+      number: '2440605262758053',
+      month: 7,
+      year: 2022,
+      verification_value: '111'
+    )
     @amount = 100
+    @sub_merchant = {
+      address: 'None',
+      extra_params: {
+      },
+      mcc: '5816',
+      merchant_id: '400000008',
+      name: '885.519.237',
+      phone_number: '3434343',
+      postal_code: 'None',
+      url: 'string'
+    }
+    @psp_info = {
+      id: '0123',
+      name: 'mci',
+      sub_merchant: {
+        id: 'string',
+        name: 'string'
+      }
 
-    setup_options()
+    }
+    @authorize_void_options_success = {
+      acquire_extra_options: {},
+      trace_id: SecureRandom.uuid,
+      user: {
+        id: '123',
+        email: 's@example.com'
+      },
+      order: {
+        id: rand(100000000000..999999999999).to_s,
+        description: 'apopsicle',
+        installments: 1,
+        datetime_local_transaction: Time.new.strftime('%Y-%m-%dT%H:%M:%S.%L%:z'),
+        amount: {
+          currency: 'USD',
+          vat: 19
+        }
+      },
+      authentication: {
+        three_ds_fields: {
+          version: '2.1.0',
+          eci: '05',
+          cavv: 'jJ81HADVRtXfCBATEp01CJUAAAA',
+          ds_transaction_id: '97267598-FAE6-48F2-8083-C23433990FBC',
+          acs_transaction_id: '13c701a3-5a88-4c45-89e9-ef65e50a8bf9',
+          cavv_algorithm: '1',
+          xid: '333333333',
+          directory_response_status: 'Y',
+          authentication_response_status: 'Y',
+          enrolled: 'test',
+          three_ds_server_trans_id: '24f701e3-9a85-4d45-89e9-af67e70d8fg8'
+        }
+      },
+      sub_merchant: @sub_merchant,
+      psp_info: @psp_info,
+      token_acquirer: @token_acquirer
+    }
+
+    @authorize_capture_options_failed = {
+      acquire_extra_options: {},
+      trace_id: SecureRandom.uuid,
+      user: {},
```

Comment:
Is the `user` object the only difference between the other `options_success` variables and this variable?

If so, it seems like we could simplify this by declaring an `@additional_options` to contain everything except `user` and then building the `success` and `failure` variables like so:
```
@authorize_options_success = {
  user: {
      id: '123',
      email: 's@example.com'
  }
}.merge(@additional_options)

@authorize_options_failed = {
  user: {}
}.merge(@additional_options)

@additional_options = {
    acquire_extra_options: {},
    trace_id: SecureRandom.uuid,
    etc...
}
```

This might make it easier to read and maintain these tests.

Reply:
That's a great idea. We have opted for implementing something very similar to this but maintaining only an `@authorize_options_success` variable and removing elements from it in the failed tests in order to malform the input data, as so: 

    options = @authorize_options_success.clone()
    options.delete(:user)

    response = @gateway.purchase(@amount, @credit_card, options)